src/mame/tandberg/tdv2100_disp_logic.cpp

```diff
@@ -0,0 +1,962 @@
+
+	PORT_START("dsw_u73")
+		PORT_DIPNAME(0x3, 0x1, "RTS signal")
+			PORT_DIPSETTING(0x1, "Automatic with DTR")                          // 1: OFF 2: ON
+			PORT_DIPSETTING(0x2, "Asserted by TRANS key")                       // 1: ON  2: OFF
+		PORT_DIPNAME(0xC, 0x8, "Current-Loop RxD line (Unused)")
+			PORT_DIPSETTING(0x4, "10V zener-diode in series")                   // 3: OFF 4: ON
+			PORT_DIPSETTING(0x8, "47 Ohm series resistor")                      // 3: ON  4: OFF
+
+	PORT_START("dsw_u61")
+		PORT_DIPNAME(0x003, 0x001, "Horizontal space between chars")
+			PORT_DIPSETTING(0x001, "Duplicate edge of previous char")           // 1: OFF 2: ON
+			PORT_DIPSETTING(0x002, "Draw gap")                                  // 1: ON  2: OFF
+		PORT_DIPNAME(0x004, 0x004, "End of line input action")
+			PORT_DIPSETTING(0x004, "Automatic CartridgeReturn + NewLine")       // 3: OFF
+			PORT_DIPSETTING(0x000, "No automatic action")                       // 3: ON
+		PORT_DIPNAME(0x008, 0x000, "Display mode")
+			PORT_DIPSETTING(0x008, "Undeline mode")                             // 4: OFF
+			PORT_DIPSETTING(0x000, "Attribute mode")                            // 4: ON
+		PORT_DIPNAME(0x030, 0x020, "Cursor behaviour")
+			PORT_DIPSETTING(0x010, "Blinking")                                  // 5: OFF 6: ON
+			PORT_DIPSETTING(0x020, "Static")                                    // 5: ON  6: OFF
+		PORT_DIPNAME(0x040, 0x000, "Attribute-spesific characters")
+			PORT_DIPSETTING(0x040, "Draw underline chars (for UL mode)")        // 7: OFF
+			PORT_DIPSETTING(0x000, "Hide attr-change chars (for Attr mode)")    // 7: ON
+		PORT_DIPNAME(0x080, 0x080, DEF_STR( Unused ))
+			PORT_DIPSETTING(0x080, DEF_STR( Off ))                              // 8: OFF
+			PORT_DIPSETTING(0x000, DEF_STR( On ))                               // 8: ON
+		PORT_DIPNAME(0x100, 0x000, "Control Characters")
+			PORT_DIPSETTING(0x100, "Draw ASCII chars 0x00 - 0x1F")              // 9: OFF
+			PORT_DIPSETTING(0x000, "Hide ASCII chars 0x00 - 0x1F")              // 9: ON
+			// NOTE: CPU module needed to write these to display RAM, as well as a ROM set with a 4th font ROM
+		PORT_DIPNAME(0x200, 0x000, "Cursor shape")
+			PORT_DIPSETTING(0x200, "Block")                                     // 10: OFF
+			PORT_DIPSETTING(0x000, "Line")                                      // 10: ON
+
+INPUT_PORTS_END
+
+ioport_constructor tandberg_tdv2100_disp_logic_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME( tdv2115l );
+}
+
+ROM_START(tdv2115l)
+	ROM_REGION( 0x0800, "display_font_rom", ROMREGION_ERASEFF )
+	ROM_LOAD( "960558-0 1.82s141.u18", 0x0200, 0x0200, CRC(002856D2) SHA1(D86AFC7190163B3C0D7A43516ADB14046B978A4F))
+	ROM_LOAD( "960558-0 2.82s141.u19", 0x0400, 0x0200, CRC(422F1959) SHA1(AAD1B74ED194C486E53D7FE6469B15C407E08441))
+	ROM_LOAD( "960558-0 3.82s141.u20", 0x0600, 0x0200, CRC(F662DF8E) SHA1(C585731612AA1BEA8D95223A49011099953CF34E))
```

Comment:
Like other hexadecimal numbers, we use lowercase digits in CRCs and SHA1 digests.

Reply:
Will fix